async-nats/src/jetstream/stream.rs

```diff
@@ -721,6 +717,127 @@ pub struct RawMessage {
     pub time: time::OffsetDateTime,
 }
 
+impl TryFrom<RawMessage> for Message {
+    type Error = Error;
+
+    fn try_from(value: RawMessage) -> Result<Self, Self::Error> {
+        let decoded_paylaod = base64::decode(value.payload)
+            .map_err(|err| Box::new(std::io::Error::new(ErrorKind::Other, err)))?;
+        let decoded_headers = {
```

Comment:
You can flatten this to map, e.g `value.headers.map(base64::decode)`

```suggestion
        let decoded_headers = {
```